domain/annotation/state/state.go

```diff
@@ -0,0 +1,292 @@
+		return nil, errors.Trace(err)
+	}
+
+	getAnnotationsStmt, err := sqlair.Prepare(getAnnotationsQuery, Annotation{}, sqlair.M{})
+	if err != nil {
+		return nil, errors.Annotatef(err, "preparing get annotations query for ID: %q", id.Name)
+	}
+
+	if id.Kind == annotations.KindModel {
+		return st.getAnnotationsForModel(ctx, db, id, getAnnotationsStmt)
+	}
+	return st.getAnnotationsForID(ctx, db, id, getAnnotationsStmt)
+}
+
+// getAnnotationsForModel retrieves all annotations aassociated with the given model id from the
+// database. If no annotations are found, an empty map is returned. This is specialized as opposed
+// to the other Kinds because we keep annotations per model, so we don't need to try to find the
+// uuid of the given id (the model).
+func (st *State) getAnnotationsForModel(ctx context.Context, db database.TxnRunner, id annotations.ID, getAnnotationsStmt *sqlair.Statement) (map[string]string, error) {
+	// Running transactions for getting annotations
+	var annotationsResults []Annotation
+	err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		return tx.Query(ctx, getAnnotationsStmt).GetAll(&annotationsResults)
+	})
+
+	if err != nil {
+		if errors.Is(err, sqlair.ErrNoRows) {
+			// No errors, we return empty map if no annotation is found
+			return map[string]string{}, nil
+		}
+		return nil, errors.Annotatef(err, "loading annotations for ID: %q", id.Name)
+	}
+
+	annotations := make(map[string]string, len(annotationsResults))
+
+	for _, result := range annotationsResults {
+		annotations[result.Key] = result.Value
+	}
+	return annotations, errors.Trace(err)
+}
+
+// getAnnotationsForID retrieves all annotations aassociated with the given id from the database. If
+// no annotations are found, an empty map is returned. This is separate from the
+// getAnnotationsForModel because for non-model ID Kinds we need to find the uuid of the id before
+// we retrieve annotations from the corresponding annotation table.
+func (st *State) getAnnotationsForID(ctx context.Context, db database.TxnRunner, id annotations.ID, getAnnotationsStmt *sqlair.Statement) (map[string]string, error) {
+	// Prepare queries for looking up the uuid of id
+	kindQuery, kindQueryParam, err := uuidQueryForID(id)
+	if err != nil {
+		return nil, errors.Annotatef(err, "preparing get annotations query for ID: %q", id.Name)
+	}
+	kindQueryStmt, err := sqlair.Prepare(kindQuery, sqlair.M{})
+	if err != nil {
+		return nil, errors.Annotatef(err, "preparing get annotations query for ID: %q", id.Name)
+	}
+
+	// Running transactions for getting annotations
+	var annotationsResults []Annotation
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		// Looking up the UUID for id
+		result := sqlair.M{}
+		err = tx.Query(ctx, kindQueryStmt, kindQueryParam).Get(result)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Annotatef(err, "looking up UUID for ID: %s", id.Name)
+		}
+
+		if len(result) == 0 {
```

Comment:
It might be more robust to check for the `sqlair.ErrNoRows` error since that should always be returned in this case.